torch/csrc/Generator.cpp

```diff
@@ -221,3 +221,38 @@ bool THPGenerator_init(PyObject *module)
   PyModule_AddObject(module, "Generator", (PyObject *)&THPGeneratorType);
   return true;
 }
+
+void set_pyobj(const Generator& self, PyObject* pyobj) {
+  TORCH_CHECK(self.defined(), "cannot call set_pyobj() on undefined generator");
+  self->set_pyobj(pyobj);
+}
+
+PyObject* pyobj(const Generator& self) {
+  TORCH_CHECK(self.defined(), "cannot call pyobj() on undefined generator");
+  return self->pyobj();
+}
+
+PyObject * THPGenerator_Wrap(Generator gen)
+{
+  if (!gen.defined()) {
+    Py_RETURN_NONE;
+  }
+
+  if (auto obj = pyobj(gen)) {
+    Py_INCREF(obj);
+    return obj;
+  }
+
+  return THPGenerator_NewWithVar((PyTypeObject *)THPGeneratorClass, std::move(gen));
+}
+
+PyObject* THPGenerator_NewWithVar(PyTypeObject* type, Generator gen)
```

Comment:
The place you copy pasted this from has a comment talking about what invariant should be upheld before calling this function; that comment should be preserved here!

Reply:
don't forget about this comment!

Reply:
I added a comment in the header, but I will add it here as well